### ISSUE TITLE
setOptions does not apply overwrite

### ISSUE THREAD

**Issue by fiws** [Automattic/mongoose#4740]
Maybe i missunderstood the docs, but this should be possible, no?

https://runkit.com/5835e1a49106c3001418f441/583747cb0ce88500131558c3

**Comment by vkarpov15** [comment]
Unfortunately that's not the way that `update()` works, `update()` casts when you call `update()` rather than when you call `exec()`, so you need to do something like this:

```javascript
const schema = new mongoose.Schema({ name: String, age: Number });
const User = mongoose.model('User', schema);

const q = User.where({ name: 'Bar' }).update({}, { name: 'Bar' }, { overwrite: true }).setOptions({ upsert: true});
```

`overwrite` is unfortunately a somewhat confusing field because what happens if you set it to true in one update and then false in another? Definitely something we want to change in the future though :+1:

**Comment by vkarpov15** [comment]
Fixed in 5.0 branch
